validation/validate.py

```diff
@@ -0,0 +1,268 @@
+    else:
+        metrics = [f'{DistanceMetric.__qualname__}.{attr}'
+                   for attr in dir(DistanceMetric)
+                   if not callable(getattr(DistanceMetric, attr))
+                   and not attr.startswith("__")]
+        err = f"Undefined {DistanceMetric.__qualname__}. Choose from {metrics}"
+        raise DistanceMetricException(err)
+
+
+def _calculate_best_threshold(thresholds: np.ndarray,
+                              dist: np.ndarray,
+                              labels: np.ndarray) -> np.float:
+    acc_train = np.zeros((len(thresholds)))
+    for threshold_idx, threshold in enumerate(thresholds):
+        predictions = np.less(dist, threshold)
+        acc_train[threshold_idx] = accuracy_score(labels, predictions)
+    best_threshold_index = np.argmax(acc_train)
+    return thresholds[best_threshold_index]
+
+
+def _score_k_fold(thresholds: np.ndarray,
+                  embeddings1: np.ndarray,
+                  embeddings2: np.ndarray,
+                  labels: np.ndarray,
+                  num_folds: int,
+                  distance_metric: DistanceMetric,
+                  subtract_mean: bool,
+                  divide_stddev: bool) -> Tuple[np.ndarray,
+                                                np.ndarray,
+                                                np.ndarray]:
+    k_fold = KFold(n_splits=num_folds, shuffle=False)
+    accuracy = np.zeros((num_folds))
+    recall = np.zeros((num_folds))
+    precision = np.zeros((num_folds))
+    splits = k_fold.split(np.arange(len(labels)))
+    for fold_idx, (train_set, test_set) in enumerate(splits):
+        train_embeddings = np.concatenate([embeddings1[train_set],
+                                           embeddings2[train_set]])
+        mean = np.mean(train_embeddings, axis=0) if subtract_mean else 0.0
+        stddev = np.std(train_embeddings, axis=0) if divide_stddev else 1.0
+        dist = _distance_between_embeddings((embeddings1 - mean) / stddev,
+                                            (embeddings2 - mean) / stddev,
+                                            distance_metric)
+        best_threshold = _calculate_best_threshold(thresholds,
+                                                   dist[train_set],
+                                                   labels[train_set])
+        predictions = np.less(dist[test_set], best_threshold)
+        accuracy[fold_idx] = accuracy_score(labels[test_set], predictions)
+        recall[fold_idx] = recall_score(labels[test_set], predictions)
+        precision[fold_idx] = precision_score(labels[test_set], predictions)
+    return accuracy, recall, precision
+
+
+def _parse_arguments():
+    parser = ArgumentParser()
+    parser.add_argument('--image_dir',
+                        type=str,
+                        required=True,
+                        help='Path to the image directory.')
+    parser.add_argument('--pairs_file_name',
+                        type=str,
```

Comment:
You can use [argparse.FileType](https://docs.python.org/3/library/argparse.html#filetype-objects) to have argparse automatically verify that the file exists for you.

Reply:
Is there a way to use FileType without it opening the file, but just verifying that it exists?